packages/dds/tree/api-report/tree.alpha.api.md

```diff
@@ -443,9 +468,34 @@ export class SchemaFactory<out TScope extends string | undefined = string | unde
     readonly string: TreeNodeSchema<"com.fluidframework.leaf.string", NodeKind.Leaf, string, string>;
 }
 
+// @alpha
+export interface SchemaValidationFunction<Schema extends TSchema> {
+    // (undocumented)
+    check(data: unknown): data is Static<Schema>;
+}
+
 // @public
 type ScopedSchemaName<TScope extends string | undefined, TName extends number | string> = TScope extends undefined ? `${TName}` : `${TScope}.${TName}`;
 
+// @alpha
+export interface SharedTreeFormatOptions {
+    formatVersion: SharedTreeFormatVersion[keyof SharedTreeFormatVersion];
+    treeEncodeType: TreeCompressionStrategy;
+}
+
+// @alpha
+export const SharedTreeFormatVersion: {
```

Comment:
Given that we default to V3 (thus dubbing it our actual "V1") and that we do not support documents in V1 (and maybe V2?), how would you feel about only exposing V3?

Reply:
My instinct is this is a foot-gun, otherwise.

Reply:
"burger king foot gungus lettuce"

Reply:
Personally I think we should just expose a single framework wide enum which lists every framework minor version which had an opt in compat impacting change, and its value of v2.0 would map to v3 in our internal format numbers.

Fixing that is mostly orthogonal to this change.

That said, if someone has legacy documents (from before 2.0), and wants to generate test documents so they can ensure their old production documents will continue to work, these extra option might be useful to them.

Since this is just alpha, I think we can sort out these details later, but I do agree this isn't want we want long term and/or when this goes public.

Reply:
Note that we haven't done the work needed to create separate public vs internal configuration, but as long as these are targeted at alpha / debugging usage I think it may be better to keep using the internal types here rather than adding all the complexity needed to create distinct public API configuration, at least for now.

I have added a TODO to the APi's private remarks. I'll let API reviewers decide if they want to block exposing these as alpha on that work.

Reply:
What happens if you set this to V1? Do we support writing legacy documents (which are immediately broken/unsupported)? I thought we only support writing N and N-1.
I'll agree that allowing V2 writing seems fine for the (possibly nonexistent) customers that want to test support for that, but exposing V1 at all seems very strange to me. @noencke thoughts?

Reply:
(regardless, I did approve and don't think this needs to block this since it's alpha)

Reply:
If we don't support V1 then I don't see why we'd expose V1, but exposing V2 makes sense for now, particularly since this is alpha.

Reply:
Should we leave some of the context from this discussion in `@privateRemarks` in the code for future reference?

Reply:
> Should we leave some of the context from this discussion in `@privateRemarks` in the code for future reference?

Is there something more you want beyond what's already in https://github.com/microsoft/FluidFramework/pull/22701/commits/fe8daec421cc22eeeb2c3e532468aeb3cacadd70 ?

Reply:
Oh, no, I didn't see that. Looks good to me!